internal/pkg/mapper/orchestrator/remote_load.go

```diff
@@ -0,0 +1,322 @@
+			continue
+		}
+		taskContent.Delete(`id`)
+
+		// Get name
+		nameRaw, found := taskContent.Get(`name`)
+		if !found {
+			errors.Append(fmt.Errorf(`missing task[%d] "name" key`, index))
+			continue
+		}
+		name, ok := nameRaw.(string)
+		if !ok {
+			errors.Append(fmt.Errorf(`task[%d] "name" must be string, found %T`, index, nameRaw))
+			continue
+		}
+		taskContent.Delete(`name`)
+
+		// Get phase
+		phaseIdRaw, found := taskContent.Get(`phase`)
+		if !found {
+			errors.Append(fmt.Errorf(`missing "phase" key in task[%d] "%s"`, index, name))
+			continue
+		}
+		phaseId, err := strconv.Atoi(cast.ToString(phaseIdRaw))
+		if err != nil {
+			errors.Append(fmt.Errorf(`task "name" key must be string, found %T, in task[%d] "%s"`, phaseIdRaw, index, name))
+			continue
+		}
+		phase, found := phaseById[cast.ToString(phaseId)]
+		if !found {
+			errors.Append(fmt.Errorf(`phase "%d" not found, referenced from task[%d] "%s"`, phaseId, index, name))
+			continue
+		}
+		taskContent.Delete(`phase`)
+
+		// Get target config
+		targetRaw, found := taskContent.Get(`task`)
+		if !found {
+			errors.Append(fmt.Errorf(`missing "task" key in task[%d] "%s"`, index, name))
+			continue
+		}
+		target, ok := targetRaw.(orderedmap.OrderedMap)
+		if !ok {
+			errors.Append(fmt.Errorf(`"task" key must be object, found %T, in task[%d] "%s"`, targetRaw, index, name))
+			continue
+		}
+
+		// Get componentId
+		componentIdRaw, found := target.Get(`componentId`)
+		if !found {
+			errors.Append(fmt.Errorf(`missing "task.componentId" key in task[%d] "%s"`, index, name))
+			continue
+		}
+		componentId, ok := componentIdRaw.(string)
+		if !ok {
+			errors.Append(fmt.Errorf(`"task.componentId" must be string, found %T, in task[%d] "%s"`, componentIdRaw, index, name))
+			continue
+		}
+		target.Delete(`componentId`)
+		taskContent.Set(`task`, target)
+
+		// Get configId
+		configIdRaw, found := target.Get(`configId`)
+		if !found {
+			errors.Append(fmt.Errorf(`missing "task.configId" key in task[%d] "%s"`, index, name))
+			continue
+		}
+		configId, ok := configIdRaw.(string)
+		if !ok {
+			errors.Append(fmt.Errorf(`"task.configId" key must be string, found %T, in task[%d] "%s"`, configIdRaw, index, name))
+			continue
+		}
+		target.Delete(`configId`)
+		taskContent.Set(`task`, target)
+
+		// Add to map
+		task := model.Task{
+			Name:        name,
+			Content:     &taskContent,
+			ComponentId: componentId,
+			ConfigId:    configId,
+		}
+		phase.Tasks = append(phase.Tasks, task)
+	}
+	return errors.ErrorOrNil()
+}
+
+func (m *orchestratorMapper) sortPhases(phaseById map[string]*model.Phase) ([]*model.Phase, error) {
+	errors := utils.NewMultiError()
+	graph := &toposort.Sorter{}
+
+	// Sort keys, so sort results will be always same.
+	// Some phases can have same deps, but their order must not be random.
+	var keys []string
+	for key := range phaseById {
+		keys = append(keys, key)
+	}
+	sort.Strings(keys)
```

Comment:
Nizsie sa pouziva `topological sort` na zoradenie podla zavislosti.
To nam ale nepomoze, ak maju viacere `phase` rovnake (alebo ziadne zavislosti),
Preto ich najprv zoradime podla ID, aby bolo vysledne poradie vzdy rovnake.